frontend/lib/resolution/disambiguation.cpp

```diff
@@ -1087,8 +690,8 @@ static void discardWorseVisibility(const DisambiguationContext& dctx,
     }
 
     DisambiguationCandidate* candidate = (DisambiguationCandidate*)candidates[i];
-
-    int distance = computeVisibilityDistance(dctx.context, dctx.callInScope, candidate->fn);
+    auto scope = dctx.callInPoiScope == nullptr ? dctx.callInScope : dctx.callInPoiScope->inScope();
```

Comment:
This seems correct based on the behavior of the production resolver with this program:

``` chapel
module A {
  proc foo(param p) {
    writeln("In A.foo");
    bar();
  }
}

module B {
  proc bar() {
    writeln("In B.bar");
  }
}

module C {
  use A;

  proc bar() {
    writeln("In C.bar");
  }

  proc main() {
    foo(1);
  }
}
```

However we should probably clarify this in the section about function visibility during disambiguation in the spec https://chapel-lang.org/docs/language/spec/procedures.html#determining-most-specific-functions

Reply:
I created https://github.com/Cray/chapel-private/issues/5707 for this task